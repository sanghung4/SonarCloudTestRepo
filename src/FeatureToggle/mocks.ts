import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
