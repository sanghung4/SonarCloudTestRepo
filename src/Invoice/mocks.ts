import { GetInvoiceDocument } from 'generated/graphql';

export const successMincron = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '123456',
      invoiceNumber: '5460026'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5460026',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: '40251/13325',
        invoiceDate: '10/28/2020',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'AVLNEP7',
        jobName: 'NEP 7 ASHEVILLE',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2020',
        deliveryMethod: 'Our Truck',
        subtotal: '3184.28',
        tax: '262.76',
        otherCharges: '0.0',
        paidToDate: '1000.00',
        invoiceItems: [
          {
            id: '1738433F4-91A7-4637-8C60-B4DF59A060C1',
            brand: 'Advanced Drainage Systems, Inc.',
            name: '120 x 7.5, 100 Sq Yd, Geosynthetic, Double, Photodegradable, Erosion Control Blanket',
            partNumber: '8X1125STRAWDN',
            mfr: '00S2TT',
            thumb: 'null',
            price: '43.59',
            qty: {
              quantityOrdered: 1,
              quantityShipped: 1
            }
          }
        ]
      }
    }
  }
};

export const successMincron_5448540 = {
  request: {
    query: GetInvoiceDocument,
    variables: {
      accountId: '91492',
      invoiceNumber: '5448540'
    }
  },
  result: {
    data: {
      invoice: {
        invoiceNumber: '5282411',
        status: 'Open',
        terms: 'NET 30 DAYS',
        customerPo: 'WILKESBORO JIM',
        invoiceDate: '10/06/2021',
        dueDate: '10/28/2020',
        originalAmt: '',
        openBalance: '',
        age: '',
        jobNumber: 'SHOP',
        jobName: 'SHOP',
        address: {
          streetLineOne: '911 BLUEBERRY WAY',
          streetLineTwo: '',
          streetLineThree: '',
          city: 'NORTHLAKE',
          state: 'TX',
          postalCode: '76226',
          country: ''
        },
        shipDate: '11/06/2021',
        deliveryMethod: 'Our Truck',
        subtotal: '2184.28',
        tax: '162.73',
        otherCharges: '0.0',
        paidToDate: '139.91',
        invoiceItems: {
          id: 'F02FD85F-4DCE-4EA7-8459-2DA7CFD689DC',
          brand: 'Certex USA, name:2 x 12, Nylon, Eye and Eye, Ply Sling',
          partNumber: 'SLING212,mfr:CX07-0126-12',
          thumb:
            'https://images.tradeservice.com/ProductImages/DIR100193/CERTEX_CX07-0126-12_LINE_SML.jpg',
          price: '40.00',
          qty: {
            quantityOrdered: 1,
            quantityShipped: 1
          }
        }
      }
    }
  }
};
