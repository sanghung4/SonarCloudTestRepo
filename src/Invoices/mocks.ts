import { subMonths } from 'date-fns';
import { Invoice, InvoicesDocument } from 'generated/graphql';
import { mockRow } from 'test-utils/mockTableInstance';
import { formatDate } from 'utils/dates';

export const success = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'ECLIPSE'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Current',
            customerPo: '1096872',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobName: null,
            jobNumber: null
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const successMincron = {
  request: {
    query: InvoicesDocument,
    variables: {
      accountId: '123456',
      erpName: 'MINCRON'
    }
  },
  result: {
    data: {
      invoices: {
        bucketNinety: 0.0,
        bucketOneTwenty: 0.0,
        bucketSixty: 0.0,
        bucketThirty: 0.0,
        bucketFuture: 0.0,
        currentAmt: 35224.96,
        invoices: [
          {
            age: 'Future',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227972.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '6566',
            jobName: 'NEP 7 ASHVELLE'
          },
          {
            age: 'Current',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227975.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: '56787',
            jobName: 'WALTER HORN'
          },
          {
            age: 'Over120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220294.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'SHOP',
            jobName: 'SEQUAYESH TR'
          },
          {
            age: '91-120',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110222345.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'RETREAT',
            jobName: 'RETREAT AT ARDEN'
          },
          {
            age: 'Deposit',
            customerPo: '109687',
            invoiceDate: formatDate(subMonths(new Date(), 1)),
            dueDate: '03/01/2022',
            invoiceNumber: 'S110220974.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Closed',
            invoiceUrl: 'test.com',
            jobNumber: 'AVLNEP7',
            jobName: 'WALTER HORN'
          },
          {
            age: '31-60',
            customerPo: '109687',
            invoiceDate: '03/01/2021',
            dueDate: '03/01/2022',
            invoiceNumber: 'S110227973.001',
            openBalance: '12.39',
            originalAmt: '12.39',
            status: 'Open',
            invoiceUrl: 'test.com',
            jobNumber: 'AIKENON',
            jobName: 'AIKEN AVL'
          }
        ],
        totalAmt: 35232.27,
        totalPastDue: 7.3085938
      }
    }
  }
};

export const mockTestTableMemo = {
  invoices: {
    bucketNinety: 0.0,
    bucketOneTwenty: 0.0,
    bucketSixty: 0.0,
    bucketThirty: 0.0,
    bucketFuture: 0.0,
    currentAmt: 35224.96,
    invoices: [
      {
        age: 'Future',
        customerPo: '109687',
        invoiceDate: '05/02/2022',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227972.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '6566',
        jobName: 'NEP 7 ASHVELLE'
      },
      {
        age: 'Current',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227975.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: '56787',
        jobName: 'WALTER HORN'
      },
      {
        age: 'Over120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220294.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'SHOP',
        jobName: 'SEQUAYESH TR'
      },
      {
        age: '91-120',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110222345.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'RETREAT',
        jobName: 'RETREAT AT ARDEN'
      },
      {
        age: 'Deposit',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110220974.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AVLNEP7',
        jobName: 'WALTER HORN'
      },
      {
        age: '31-60',
        customerPo: '109687',
        invoiceDate: '03/01/2021',
        dueDate: '03/01/2022',
        invoiceNumber: 'S110227973.001',
        openBalance: '12.39',
        originalAmt: '12.39',
        status: 'Open',
        invoiceUrl: 'test.com',
        jobNumber: 'AIKENON',
        jobName: 'AIKEN AVL'
      }
    ],
    totalAmt: 35232.27,
    totalPastDue: 7.3085938
  }
};

const rowToMock = success.result.data.invoices.invoices[0];
export const mockInvoiceRow = mockRow<Invoice>(rowToMock, rowToMock, {});
