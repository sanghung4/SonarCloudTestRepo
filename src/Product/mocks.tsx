import {
  GetProductDocument,
  GetProductPricingDocument
} from 'generated/graphql';

export const success = {
  request: {
    query: GetProductDocument,
    variables: {
      productInput: { productId: 'MSC-33860', customerNumber: undefined }
    }
  },
  result: {
    data: {
      product: {
        id: 'MSC-33860',
        partNumber: '33860',
        name: '1/2" Wrot Copper Cap',
        manufacturerName: 'Generic',
        manufacturerNumber: 'CCAPD',
        price: 0,
        stock: null,
        taxonomy: null,
        categories: [
          'Pipe & Fittings',
          'Metal Pipe & Fittings',
          'Copper Tube & Fittings'
        ],
        technicalDocuments: [
          {
            name: 'MFR Catalog Page',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00056_18_20.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'Technical Specification',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100240/MUELERE03088_1.pdf',
            __typename: 'TechDoc'
          },
          {
            name: 'MFR Item Data',
            url: 'http://images.tradeservice.com/ATTACHMENTS/DIR100145/MUELERE00601_1.pdf',
            __typename: 'TechDoc'
          }
        ],
        environmentalOptions: [],
        upc: '68576823241',
        unspsc: '40183104',
        seriesModelFigureNumber: null,
        productOverview:
          'Cap Fitting; Type Round Head; Nominal Size 1/2"; End Connection Female Soldered; Material Copper; Process Wrot; Application Ground Potable Water Supply System; Applicable Standard ASME B16.22, ANSI/NSF 61 G; Packaging Quantity 50 per Inner Pack, 1000 per Master Carton',
        featuresAndBenefits:
          'Copper Solder-Joint Fittings for Supply/Pressurized Systems have Been the Leading Brand of Copper Fittings for Over 80 Years; Available in Both Wrot Copper and Cast Bronze; The Acknowledged Experts at Engineering and Manufacturing Precision Solder-Joint Copper Fittings; Quality, Consistency and Reliability have Made the Streamline Brand Trusted and Specified All Around the World',
        techSpecifications: [
          {
            name: 'material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Applicable Standard',
            value: 'ASME B16.22, ANSI/NSF 61 G',
            __typename: 'TechSpec'
          },
          {
            name: 'Application',
            value: 'Ground Potable Water Supply System',
            __typename: 'TechSpec'
          },
          {
            name: 'End Connection',
            value: 'Female Soldered',
            __typename: 'TechSpec'
          },
          {
            name: 'Material',
            value: 'Copper',
            __typename: 'TechSpec'
          },
          {
            name: 'Nominal Size',
            value: '1/2"',
            __typename: 'TechSpec'
          },
          {
            name: 'Process',
            value: 'Wrot',
            __typename: 'TechSpec'
          },
          {
            name: 'Type',
            value: 'Round Head',
            __typename: 'TechSpec'
          }
        ],
        imageUrls: {
          thumb:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          small:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_SML.jpg',
          medium:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_MED.jpg',
          large:
            'http://images.tradeservice.com/ProductImages/DIR100103/MUELER_W-07004_LRG.jpg',
          __typename: 'ImageUrls'
        },
        packageDimensions: null,
        minIncrementQty: 0,
        erp: 'ECLIPSE',
        __typename: 'Product'
      }
    }
  }
};

export const pricing = {
  request: {
    query: GetProductPricingDocument,
    variables: {
      input: {
        customerId: '',
        branchId: 'any',
        productIds: ['33860'],
        includeListData: false
      }
    }
  },
  result: {
    data: {
      productPricing: {
        customerId: 'testAccount',
        branch: 'any',
        products: [
          {
            productId: '33860',
            orderUom: 'ea',
            catalogId: '',
            branchAvailableQty: 2,
            totalAvailableQty: 3,
            listIds: [],
            sellPrice: 3.256
          }
        ]
      }
    }
  }
};
