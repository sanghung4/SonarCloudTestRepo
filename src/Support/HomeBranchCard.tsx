import React from 'react';

import {
  Box,
  Card,
  CardHeader,
  Link,
  Skeleton,
  Typography,
  useScreenSize
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';

import { useGetHomeBranchQuery } from 'generated/graphql';
import homeBranchImg from 'images/home-branch.svg';
import { useSelectedAccountsContext } from 'providers/SelectedAccountsProvider';

function HomeBranchCard() {
  /**
   * Custom Hooks
   */
  const { isSmallScreen } = useScreenSize();
  const { t } = useTranslation();

  /**
   * Context
   */
  const { selectedAccounts } = useSelectedAccountsContext();

  /**
   * Data
   */
  const { data: homeBranchQuery, loading: homeBranchLoading } =
    useGetHomeBranchQuery({
      variables: {
        shipToAccountId: selectedAccounts?.shipTo?.id ?? ''
      }
    });

  return !homeBranchQuery?.homeBranch && homeBranchLoading ? (
    <Box
      width={isSmallScreen ? '21.4rem' : '32.4rem'}
      height={isSmallScreen ? '14.6rem' : '8.4rem'}
    >
      <Card>
        <Box
          display="flex"
          sx={
            isSmallScreen
              ? { flexDirection: 'column', alignItems: 'center' }
              : {}
          }
        >
          <Box p={3}>
            <Skeleton variant="rectangular" height={80} width={80} />
          </Box>
          <Box
            display={isSmallScreen ? 'flex' : ''}
            sx={
              isSmallScreen
                ? { flexDirection: 'column', alignItems: 'center' }
                : {}
            }
          >
            <Typography variant="h2" component="span">
              <Skeleton width={250} />
            </Typography>

            <Typography variant="h4" component="span">
              <Skeleton width={200} />
            </Typography>

            <Typography variant="h4" component="span">
              <Skeleton width={200} />
            </Typography>
          </Box>
        </Box>
      </Card>
    </Box>
  ) : (
    <Card sx={{ borderRadius: isSmallScreen ? '.5rem' : undefined }}>
      <Box p={3} display={isSmallScreen ? 'block' : 'flex'}>
        <Box>
          <Box
            component="img"
            alt="HomeBranch"
            src={homeBranchImg}
            sx={{
              mx: 'auto',
              display: 'block'
            }}
          />
        </Box>
        <Box>
          <Box mt={-2} mb={-1.5}>
            <CardHeader
              align={isSmallScreen ? 'center' : ''}
              title={homeBranchQuery?.homeBranch?.name ?? ''}
              titleTypographyProps={{
                variant: 'h4',
                color: 'primary'
              }}
            />
          </Box>
          <Box px={3} textAlign={isSmallScreen ? 'center' : undefined}>
            <Typography variant="body2" color="primary">
              {homeBranchQuery?.homeBranch?.address1 ?? ''}{' '}
              {homeBranchQuery?.homeBranch?.address2 ?? ''}{' '}
              {homeBranchQuery?.homeBranch?.city ?? ''},{' '}
              {homeBranchQuery?.homeBranch?.state ?? ''}{' '}
              {homeBranchQuery?.homeBranch?.zip ?? ''}
            </Typography>
          </Box>
          {homeBranchQuery?.homeBranch?.phone ? (
            <Box
              px={3}
              justifyContent={isSmallScreen ? 'center' : ''}
              display="flex"
              mt={1.5}
            >
              <Typography
                variant="body2"
                color="textSecondary"
                component="span"
              >
                {t('common.phoneNumber')}:{' '}
              </Typography>
              <Link
                href={`tel:${homeBranchQuery?.homeBranch?.phone}`}
                color="primary02.main"
                sx={{ ml: 0.5, textDecoration: 'none' }}
              >
                {homeBranchQuery?.homeBranch?.phone}
              </Link>
            </Box>
          ) : null}
        </Box>
      </Box>
    </Card>
  );
}

export default HomeBranchCard;
