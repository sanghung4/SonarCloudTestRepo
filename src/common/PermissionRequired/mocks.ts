import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
