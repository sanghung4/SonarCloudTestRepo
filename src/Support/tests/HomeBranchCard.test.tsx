import HomeBranchCard from 'Support/HomeBranchCard';
import { setBreakpoint } from 'test-utils/mockMediaQuery';
import { render } from 'test-utils/TestWrapper';
import { HomeBranchCardMocks } from 'Support/tests/mocks';
import { waitFor } from '@testing-library/react';

const shipToId = HomeBranchCardMocks[0].request.variables.shipToAccountId;

describe('Support - Home Branch Card', () => {
  it('should match snapshot on desktop', async () => {
    setBreakpoint('desktop');
    const { container } = render(<HomeBranchCard />, {
      mocks: [HomeBranchCardMocks[0]],
      selectedAccountsConfig: { selectedAccounts: { shipTo: { id: shipToId } } }
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 1)));
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on mobile', async () => {
    setBreakpoint('mobile');
    const { container } = render(<HomeBranchCard />, {
      mocks: [HomeBranchCardMocks[0]],
      selectedAccountsConfig: { selectedAccounts: { shipTo: { id: shipToId } } }
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 1)));
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on desktop with no Home Branch', () => {
    setBreakpoint('desktop');
    const { container } = render(<HomeBranchCard />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on mobile with no Home Branch', () => {
    setBreakpoint('mobile');
    const { container } = render(<HomeBranchCard />);
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on desktop with empty data', async () => {
    setBreakpoint('desktop');
    const { container } = render(<HomeBranchCard />, {
      mocks: [HomeBranchCardMocks[1]],
      selectedAccountsConfig: { selectedAccounts: { shipTo: { id: shipToId } } }
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 1)));
    expect(container).toMatchSnapshot();
  });

  it('should match snapshot on mobile with empty data', async () => {
    setBreakpoint('mobile');
    const { container } = render(<HomeBranchCard />, {
      mocks: [HomeBranchCardMocks[1]],
      selectedAccountsConfig: { selectedAccounts: { shipTo: { id: shipToId } } }
    });
    await waitFor(() => new Promise((res) => setTimeout(res, 1)));
    expect(container).toMatchSnapshot();
  });
});
