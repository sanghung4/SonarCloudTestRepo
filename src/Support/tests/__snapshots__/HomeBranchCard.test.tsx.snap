// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Support - Home Branch Card should match snapshot on desktop 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gs6in-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-1588wxq"
    >
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="HomeBranch"
          class="MuiBox-root css-1t9djzq"
          src="home-branch.svg"
        />
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1wppjvp"
        >
          <div
            align=""
            class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-x80elu-MuiTypography-root"
              >
                Fortiline Waterworks
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-t0051x"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1mcg7za-MuiTypography-root"
          >
            1628 Barton Chapel Road
             
            
             
            Augusta
            ,
             
            Georgia
             
            30909
          </p>
        </div>
        <div
          class="MuiBox-root css-1yfxu03"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 css-1yo1z97-MuiTypography-root"
          >
            Phone Number
            :
             
          </span>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1uzpfpa-MuiTypography-root-MuiLink-root"
            href="tel:(703) 737-3705"
          >
            (703) 737-3705
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Support - Home Branch Card should match snapshot on desktop with empty data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gs6in-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-1588wxq"
    >
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="HomeBranch"
          class="MuiBox-root css-1t9djzq"
          src="home-branch.svg"
        />
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1wppjvp"
        >
          <div
            align=""
            class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-x80elu-MuiTypography-root"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-t0051x"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1mcg7za-MuiTypography-root"
          >
            
             
            
             
            
            ,
             
            
             
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Support - Home Branch Card should match snapshot on desktop with no Home Branch 1`] = `
<div>
  <div
    class="MuiBox-root css-12qhuyy"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gs6in-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-k008qs"
      >
        <div
          class="MuiBox-root css-19kzrtu"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
            style="width: 80px; height: 80px;"
          />
        </div>
        <div
          class="MuiBox-root css-1rzdnss"
        >
          <span
            class="MuiTypography-root MuiTypography-h2 css-llmy31-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 250px;"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-h4 css-rbn354-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 200px;"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-h4 css-rbn354-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 200px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Support - Home Branch Card should match snapshot on mobile 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-17dmvp0-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-1voojg0"
    >
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="HomeBranch"
          class="MuiBox-root css-1t9djzq"
          src="home-branch.svg"
        />
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1wppjvp"
        >
          <div
            align="center"
            class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-x80elu-MuiTypography-root"
              >
                Fortiline Waterworks
              </span>
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-3ciuqz"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1mcg7za-MuiTypography-root"
          >
            1628 Barton Chapel Road
             
            
             
            Augusta
            ,
             
            Georgia
             
            30909
          </p>
        </div>
        <div
          class="MuiBox-root css-cr91pz"
        >
          <span
            class="MuiTypography-root MuiTypography-body2 css-1yo1z97-MuiTypography-root"
          >
            Phone Number
            :
             
          </span>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-1uzpfpa-MuiTypography-root-MuiLink-root"
            href="tel:(703) 737-3705"
          >
            (703) 737-3705
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Support - Home Branch Card should match snapshot on mobile with empty data 1`] = `
<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-17dmvp0-MuiPaper-root-MuiCard-root"
  >
    <div
      class="MuiBox-root css-1voojg0"
    >
      <div
        class="MuiBox-root css-0"
      >
        <img
          alt="HomeBranch"
          class="MuiBox-root css-1t9djzq"
          src="home-branch.svg"
        />
      </div>
      <div
        class="MuiBox-root css-0"
      >
        <div
          class="MuiBox-root css-1wppjvp"
        >
          <div
            align="center"
            class="MuiCardHeader-root css-1bnehoc-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h4 MuiCardHeader-title css-x80elu-MuiTypography-root"
              />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-3ciuqz"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-1mcg7za-MuiTypography-root"
          >
            
             
            
             
            
            ,
             
            
             
            
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Support - Home Branch Card should match snapshot on mobile with no Home Branch 1`] = `
<div>
  <div
    class="MuiBox-root css-wc6la7"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-13gs6in-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiBox-root css-dvxtzn"
      >
        <div
          class="MuiBox-root css-19kzrtu"
        >
          <span
            class="MuiSkeleton-root MuiSkeleton-rectangular MuiSkeleton-pulse css-3x7rqn-MuiSkeleton-root"
            style="width: 80px; height: 80px;"
          />
        </div>
        <div
          class="MuiBox-root css-dvxtzn"
        >
          <span
            class="MuiTypography-root MuiTypography-h2 css-llmy31-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 250px;"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-h4 css-rbn354-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 200px;"
            />
          </span>
          <span
            class="MuiTypography-root MuiTypography-h4 css-rbn354-MuiTypography-root"
          >
            <span
              class="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-pulse css-18tyrrx-MuiSkeleton-root"
              style="width: 200px;"
            />
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
